### ISSUE TITLE
JWT::DecodeError Nil JSON web token when connecting to a Xero account

### ISSUE THREAD

**Issue by gdelavil** [XeroAPI/xero-oauth2-omniauth-strategy#3]
### Background
When trying to connect to a new account, the server fails on callback with an error `JWT::DecodeError Nil JSON web token when connecting to a Xero account` . I thought this could've been an implementation issue on my end, but I've tried the demo app and I get the same.  Gem version is 0.9.2.

### Steps to reproduce

- Clone the demo app
- Replace the credentials with your own
- Install and start the app as per the repo's instructions
- Click the "Connect to Xero" button and give all of the approvals

### Expected result
Come back to the app with a new user (or at least a successful result

### What actually happens
We get an error screen on the callback phase. Relevant log:

```
INFO -- omniauth: (xero_oauth2) Callback phase initiated.
  
JWT::DecodeError (Nil JSON web token):
  
.bundle/gems/jwt-2.2.1/lib/jwt/decode.rb:12:in `initialize'
.bundle/gems/jwt-2.2.1/lib/jwt.rb:28:in `new'
.bundle/gems/jwt-2.2.1/lib/jwt.rb:28:in `decode'
.bundle/gems/omniauth-xero-oauth2-0.9.2/lib/omniauth/strategies/xero_oauth2.rb:55:in `raw_info'
.bundle/gems/omniauth-xero-oauth2-0.9.2/lib/omniauth/strategies/xero_oauth2.rb:39:in `block in <class:XeroOauth2>'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:109:in `instance_eval'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:109:in `block in compile_stack'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:108:in `each'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:108:in `inject'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:108:in `compile_stack'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:102:in `uid_stack'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:332:in `uid'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:348:in `auth_hash'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:372:in `callback_phase'
.bundle/gems/omniauth-oauth2-1.6.0/lib/omniauth/strategies/oauth2.rb:75:in `callback_phase'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:238:in `callback_call'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:189:in `call!'
.bundle/gems/omniauth-1.9.0/lib/omniauth/strategy.rb:169:in `call'
.bundle/gems/omniauth-1.9.0/lib/omniauth/builder.rb:64:in `call'
.bundle/gems/rack-2.0.7/lib/rack/tempfile_reaper.rb:15:in `call'
.bundle/gems/rack-2.0.7/lib/rack/etag.rb:25:in `call'
.bundle/gems/rack-2.0.7/lib/rack/conditional_get.rb:25:in `call'
.bundle/gems/rack-2.0.7/lib/rack/head.rb:12:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
.bundle/gems/rack-2.0.7/lib/rack/session/abstract/id.rb:232:in `context'
.bundle/gems/rack-2.0.7/lib/rack/session/abstract/id.rb:226:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/cookies.rb:670:in `call'
.bundle/gems/activerecord-5.2.3/lib/active_record/migration.rb:559:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/callbacks.rb:28:in `block in call'
.bundle/gems/activesupport-5.2.3/lib/active_support/callbacks.rb:98:in `run_callbacks'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/executor.rb:14:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/debug_exceptions.rb:61:in `call'
.bundle/gems/web-console-3.7.0/lib/web_console/middleware.rb:135:in `call_app'
.bundle/gems/web-console-3.7.0/lib/web_console/middleware.rb:30:in `block in call'
.bundle/gems/web-console-3.7.0/lib/web_console/middleware.rb:20:in `catch'
.bundle/gems/web-console-3.7.0/lib/web_console/middleware.rb:20:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
.bundle/gems/railties-5.2.3/lib/rails/rack/logger.rb:38:in `call_app'
.bundle/gems/railties-5.2.3/lib/rails/rack/logger.rb:26:in `block in call'
.bundle/gems/activesupport-5.2.3/lib/active_support/tagged_logging.rb:71:in `block in tagged'
.bundle/gems/activesupport-5.2.3/lib/active_support/tagged_logging.rb:28:in `tagged'
.bundle/gems/activesupport-5.2.3/lib/active_support/tagged_logging.rb:71:in `tagged'
.bundle/gems/railties-5.2.3/lib/rails/rack/logger.rb:26:in `call'
.bundle/gems/sprockets-rails-3.2.1/lib/sprockets/rails/quiet_assets.rb:13:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/request_id.rb:27:in `call'
.bundle/gems/rack-2.0.7/lib/rack/method_override.rb:22:in `call'
.bundle/gems/rack-2.0.7/lib/rack/runtime.rb:22:in `call'
.bundle/gems/activesupport-5.2.3/lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/executor.rb:14:in `call'
.bundle/gems/actionpack-5.2.3/lib/action_dispatch/middleware/static.rb:127:in `call'
.bundle/gems/rack-2.0.7/lib/rack/sendfile.rb:111:in `call'
.bundle/gems/railties-5.2.3/lib/rails/engine.rb:524:in `call'
.bundle/gems/puma-3.12.1/lib/puma/configuration.rb:227:in `call'
.bundle/gems/puma-3.12.1/lib/puma/server.rb:660:in `handle_request'
.bundle/gems/puma-3.12.1/lib/puma/server.rb:474:in `process_client'
.bundle/gems/puma-3.12.1/lib/puma/server.rb:334:in `block in run'
.bundle/gems/puma-3.12.1/lib/puma/thread_pool.rb:135:in `block in spawn_thread'
```

### What I've tried and worked
On line 55, replace `@id_token` with `id_token`:

`decoded_info ||= JWT.decode id_token, nil, false`

Please let me know if you'd rather I opened a PR with that change. I'm not sure if this is the fix you'd be thinking of, or if `id_token` is meant to be running elsewhere and not doing so.

**Comment by jenksguo** [comment]
Hi @gdelavil , thanks for raising the issue and creating the pull request. 

This is an oversight on the strategy, the @id_token should not be used for JWT.decode method as it will be nil when the scope does not include openID. 

While the id_token might work, I have however decided to use the omniauth auth_hash access_token['id_token'] to be consistent  with the evaluation statement of the code block:

        if **access_token['id_token']** == nil
          @raw_info = {
            'xero_userid'=> '',
            'given_name' => '',
            'family_name' => '',
            'email' => '',
          }
        else
          decoded_info ||= JWT.decode **access_token['id_token']**, nil, false
          @raw_info ||= decoded_info[0]
        end

I have published the fix in a new version, please use 0.9.3. 

In the future, you can create a branch and submit a pull request directly on this repo.

**Comment by gdelavil** [comment]
Thanks @jenksguo ! Yeah, I made a fork following the Xero guidelines, but I'm happy to create a branch on this repo if you are.
